ZAPD/ZTexture.cpp

```diff
@@ -756,17 +797,39 @@ Declaration* ZTexture::DeclareVar(const std::string& prefix,
 std::string ZTexture::GetBodySourceCode() const
 {
 	std::string sourceOutput;
+	std::string typeSizeStr =
+		Globals::Instance->outputPath.stem().stem().stem().extension().c_str();
+	TextureTypeSize texTypeSize = TextureTypeSize::Size64;
 
-	for (size_t i = 0; i < textureDataRaw.size(); i += 8)
+	if (!typeSizeStr.empty())
 	{
-		if (i % 32 == 0)
-			sourceOutput += "    ";
-
-		sourceOutput +=
-			StringHelper::Sprintf("0x%016llX, ", BitConverter::ToUInt64BE(textureDataRaw, i));
+		texTypeSize = static_cast<TextureTypeSize>(StringHelper::StrToL(typeSizeStr.substr(2)));
```

Comment:
This all assumes that the file is named in the correct format, which can't be assumed for anything not made by ZAPD, and even with ZAPD: something as simple as putting an extra `.` in the filename will break this.

Reply:
Yeah, that was how I originally did it. You have to add a few lines to the makefile, but it's not a big deal. And then, ZAPD can error on an invalid name before it even gets to the processing (although it needs the size of the data to test for an incompatible type).

Reply:
Discussion moved to 
https://discord.com/channels/688807550715560050/793267277859520563/897511296021438474